semgrep-core/src/core/Semgrep_error_code.ml

```diff
@@ -106,13 +106,14 @@ let exn_to_error ?(rule_id = None) file exn =
   | UnixExit _ as exn -> raise exn
   (* general case, can't extract line information from it, default to line 1 *)
   | exn ->
+      let trace = Printexc.get_backtrace () in
       let loc = Parse_info.first_loc_of_file file in
       {
         rule_id;
         typ = FatalError;
         loc;
         msg = Common.exn_to_s exn;
-        details = Some (Printexc.get_backtrace ());
+        details = Some trace;
         yaml_path = None;
       }
```

Comment:
For those interested in ocaml, what I do here is capture the backtrace as soon as possible after catching the exception. It actually would be best to capture it in the `with` part of the original try-with. The problem is that calling a function before capturing the backtrace can clear the backtrace. This happens if such a function raises and catches exceptions internally.

Reply:
Ah, makes sense. I should have realized that myself. Maybe we can write a semgrep rule for this